src/access/AccessControl.cpp

```diff
@@ -174,8 +171,17 @@ char GetPrivilegeStringForLogging(Privilege privilege)
 
 } // namespace
 
-AccessControl::Entry::Delegate AccessControl::Entry::mDefaultDelegate;
-AccessControl::EntryIterator::Delegate AccessControl::EntryIterator::mDefaultDelegate;
+AccessControl::Entry::Delegate & AccessControl::Entry::GetDefaultDelegate()
+{
+    static Delegate instance;
```

Comment:
Note that this has a code size and runtime penalty compared to the previous version because C++ requires that static local variable initialization is thread-safe: https://en.cppreference.com/w/cpp/language/storage_duration#Static_local_variables. Do you know if the increase reported by bloatcheck if because of that or because of the changes in Optional class? In the first case, you could consider introducing a helper class for controlling initialization of global objects on different platforms.

```c++
#if CHIP_CONFIG_GLOBAL_LAZY_INITIALIZATION

template <class T>
class Global
{
public:
    T& operator*()
    {
      if (!mObject.HasValue())
      {
        mObject.Emplace();
      }
      
      return mObject;
    }

private:
    Optional<T> mObject;
};

#else

template <class T>
class Global
{
public:
    T& operator*() { return *mObject; }

private:
    T* mObject;
};
#endif
```

Reply:
I'll split this up into separate commits to see if the Optional change is affecting code size negatively by itself (I've also fixed it to use a 0-sized `std::monostate` instead of a dummy char for the empty case).

A helper class like what you've sketched out might be the way to go for making globals optionally lazy.

Reply:
I've implemented something like this approach now with a `chip::Global` helper.